src/docs/swift-code-generation.md

```diff
@@ -0,0 +1,70 @@
+# Swift code generation
+
+Certain parts of the Swift ABI are unsable, such as metadata, async context, dynamic dispatch thunks, and cannot be directly projected into .NET without Swift wrappers. These wrappers provide the flexibility to encapsulate unstable ABI parts into a stable calling convention.
```

Comment:
How can Apple use these in Apple OS APIs when they are unstable?

Reply:
Referring to runtime-generated features, such as vtable in the case of generics. Additionally, this would require reverse-engineering async and closure contexts.

It's likely we'll encounter cases that are hard to represent directly in C#. Even if we manage to do so, the maintenance cost would be high. Swift wrappers can address this issue by leveraging the Swift compiler/runtime to bridge the gap.

Reply:
I do not think "unstable ABI" is the right word to describe this. Unstable ABI is expected to break from version to version. OS APIs that apps depend on cannot have unstable ABI - it holds independent on .NET. 

I would call this "ABI details that are too complex to project to .NET".

Reply:
Alternative: "poorly documented ABI details that are too complex to project to .NET".